mlflow/gateway/config.py

```diff
@@ -44,18 +44,66 @@ def validate_api_key(cls, value):
         return _resolve_api_key_from_input(value)
 
 
+class OpenAIAPIType(str, Enum):
+    OPENAI = "openai"
+    AZURE = "azure"
+    AZUREAD = "azuread"
+
+    @classmethod
+    def _missing_(cls, value):
+        """
+        Implements case-insensitive matching of API type strings
+        """
+        for api_type in cls:
+            if api_type.value == value.lower():
+                return api_type
+
+
 class OpenAIConfig(BaseModel, extra=Extra.allow):
     openai_api_key: str
-    openai_api_type: Optional[str] = None
-    openai_api_base: str = "https://api.openai.com/v1"
+    openai_api_type: OpenAIAPIType = OpenAIAPIType.OPENAI
+    openai_api_base: Optional[str] = None
     openai_api_version: Optional[str] = None
+    openai_deployment_name: Optional[str] = None
     openai_organization: Optional[str] = None
 
     # pylint: disable=no-self-argument
     @validator("openai_api_key", pre=True)
     def validate_openai_api_key(cls, value):
         return _resolve_api_key_from_input(value)
 
+    @root_validator(pre=False)
+    def validate_field_compatibility(cls, config: "OpenAIConfig"):
```

Comment:
I think the type of `config` is dict.

Reply:
Indeed it is - fixed! Would be great to enable linting on type hints